tutorials/0_What_is_ROS.ipynb

```diff
@@ -0,0 +1,150 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "toc": true
+   },
+   "source": [
+    "<h1>Table of Contents<span class=\"tocSkip\"></span></h1>\n",
+    "<div class=\"toc\"><ul class=\"toc-item\"><li><span><a href=\"#What-is-ROS-?\" data-toc-modified-id=\"What-is-ROS-?-1\"><span class=\"toc-item-num\">1&nbsp;&nbsp;</span>What is ROS ?</a></span></li><li><span><a href=\"#How-does-ROS-work-?\" data-toc-modified-id=\"How-does-ROS-work-?-2\"><span class=\"toc-item-num\">2&nbsp;&nbsp;</span>How does ROS work ?</a></span><ul class=\"toc-item\"><li><span><a href=\"#ROS-master\" data-toc-modified-id=\"ROS-master-2.1\"><span class=\"toc-item-num\">2.1&nbsp;&nbsp;</span>ROS master</a></span></li></ul></li><li><span><a href=\"#Why-ROS-?\" data-toc-modified-id=\"Why-ROS-?-3\"><span class=\"toc-item-num\">3&nbsp;&nbsp;</span>Why ROS ?</a></span></li><li><span><a href=\"#Sources:\" data-toc-modified-id=\"Sources:-4\"><span class=\"toc-item-num\">4&nbsp;&nbsp;</span>Sources:</a></span></li></ul></div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<img src=\"images/ros_logo.png\" width=\"200\" />\n",
+    "\n",
+    "# What is ROS ?\n",
+    "\n",
+    "ROS is an abbreviation for '*Robot Operating System*' and is **open source** robotics middleware, which means it is a collection of software frameworks for robotic software development. It provides the services you would expect from an operating system, including hardware abstraction, low-level device control, implementation of commonly-used functionality, message-passing between processes, and package management which can be used to create robot applications.\n",
+    "\n",
+    "The primary goal of ROS is to support the reuse of code. ROS is a distributed framework of processes that enables executables to be individually designed and loosely coupled. These processes are called **Nodes** and can be grouped into **Packages** and **Stacks**. This makes sharing and distributing code easy. ROS also supports a system of code **Repositories** to collaborate on projects and distribute them. Github is commonly used for these collaborations. This allows you to more easily manage and maintain your projects.\n",
```

Comment:
ROS **does not** support repositories. Users _host_ code on repositories like GitHub but they might as well distribute code by an archive file hosted on a webserver running on their personal computer or send the code printed on paper to the end user who types it into the system. GitHub is first and foremost a version control system! So I think saying ROS supports repositories is very misleading.